internal/pkg/manifest/svc.go

```diff
@@ -285,6 +285,29 @@ func (a *AdvancedCount) hasScalingFieldsSet() bool {
 	}
 }
 
+func (a *AdvancedCount) getInvalidFieldsSet() []string {
```

Comment:
Can we do this in the other way around using `validScalingFields()`? So that by default we throw incompatible error unless it is specifically defined as a valid one for this type of workload.

This can be a good reminder if we forget to update this function if we add more supports for some scaling policies in the future.

Reply:
That would certainly simplify it. I made it like this because I wanted it to respond by saying exactly what scaling fields that are set are invalid, but users have the resources and are likely able to figure this out regardless. So in hindsight I agree that it's worth making the code simpler to have a slightly less informative error message. Will change 👍

Reply:
Though looking more at it, how should we store all the invalid scaling fields? There's a list of which ones are valid for each type, but not a comprehensive list of scaling fields in total. Should I make a constant array of them to compare against?

Reply:
Yeah i think also the reminder thing is what I am mostly interested in too. Because if we forget to update `getInvalidFieldsSet` then most likely users will not be aware of the issue whereas if we do it the other way around, before shipping this feature we would know because it errors out